internal/fs/fs.go

```diff
@@ -246,10 +246,16 @@ func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHand
 
 	// When user passes allow_other flag, then other users should be able to
 	// read from cache
+	isPermissible, err := util.CheckCacheDirectoryPermissions(cacheLocation)
```

Comment:
This code change should be before allow_other comment?

Reply:
done in latest commit